salt/cloud/clouds/opennebula.py

```diff
@@ -552,47 +645,97 @@ def _get_node(name):
                     name, attempts
                 )
             )
+
             # Just a little delay between attempts...
             time.sleep(0.5)
+
     return {}
 
 
-def destroy(name, call=None):
+def _get_xml_rpc():
     '''
-    Destroy a node. Will check termination protection and warn if enabled.
+    Uses the OpenNebula cloud provider configurations to connect to the
+    OpenNebula API.
 
-    CLI Example:
+    Returns the server connection created as well as the user and password
+    values from the cloud provider config file used to make the connection.
+    '''
+    vm_ = get_configured_provider()
 
-    .. code-block:: bash
+    xml_rpc = config.get_cloud_config_value(
+        'xml_rpc', vm_, __opts__, search_global=False
+    )
 
-        salt-cloud --destroy mymachine
-    '''
-    if call == 'function':
-        raise SaltCloudSystemExit(
-            'The destroy action must be called with -d, --destroy, '
-            '-a or --action.'
-        )
+    user = config.get_cloud_config_value(
+        'user', vm_, __opts__, search_global=False
+    )
 
-    salt.utils.cloud.fire_event(
-        'event',
-        'destroying instance',
-        'salt/cloud/{0}/destroying'.format(name),
-        {'name': name},
+    password = config.get_cloud_config_value(
+        'password', vm_, __opts__, search_global=False
     )
 
+    server = salt.ext.six.moves.xmlrpc_client.ServerProxy(xml_rpc)
+
+    return server, user, password
+
+
+def _list_nodes(full=False):
+    '''
+    Helper function for the list_* query functions - Constructs the
+    appropriate dictionaries to return from the API query.
+
+    full
+        If performing a full query, such as in list_nodes_full, change
+        this parameter to ``True``.
+    '''
     server, user, password = _get_xml_rpc()
+    auth = user+':'+password
```

Comment:
Oh, yeah that's better. I meant to update those but forgot. I'll fix those references now.